spec/directives/for/basic.hrx

```diff
@@ -0,0 +1,145 @@
+<===> inclusive_forward/input.scss
+a {
+  @for $i from 1 through 5 {b: $i;}
+}
+
+<===> inclusive_forward/output.css
+a {
+  b: 1;
+  b: 2;
+  b: 3;
+  b: 4;
+  b: 5;
+}
+
+<===>
+================================================================================
+<===> inclusive_backward/input.scss
+a {
+  @for $i from 5 through 1 {b: $i;}
+}
+
+<===> inclusive_backward/output.css
+a {
+  b: 5;
+  b: 4;
+  b: 3;
+  b: 2;
+  b: 1;
+}
+
+<===>
+================================================================================
+<===> empty/input.scss
+a {
+  @for $i from 1 to 1 {b: $i;}
+}
+
+<===> empty/output.css
+
+<===>
+================================================================================
+<===> exclusive_forward/input.scss
+a {
+  @for $i from 1 to 5 {b: $i;}
+}
+
+<===> exclusive_forward/output.css
+a {
+  b: 1;
+  b: 2;
+  b: 3;
+  b: 4;
+}
+
+<===>
+================================================================================
+<===> exclusive_backward/input.scss
+a {
+  @for $i from 5 to 1 {b: $i;}
+}
+
+<===> exclusive_backward/output.css
+a {
+  b: 5;
+  b: 4;
+  b: 3;
+  b: 2;
+}
+
+<===>
+================================================================================
+<===> negative_to_positive/input.scss
+a {
+  @for $i from -1 through 3 {b: $i;}
+}
+
+<===> negative_to_positive/output.css
+a {
+  b: -1;
+  b: 0;
+  b: 1;
+  b: 2;
+  b: 3;
+}
+
+<===>
+================================================================================
+<===> positive_to_negative/input.scss
+a {
+  @for $i from 2 through -1 {b: $i;}
+}
+
+<===> positive_to_negative/output.css
+a {
+  b: 2;
+  b: 1;
+  b: 0;
+  b: -1;
+}
+
+<===>
+================================================================================
+<===> negative_to_negative/input.scss
+a {
+  @for $i from -5 through -1 {b: $i;}
+}
+
+<===> negative_to_negative/output.css
+a {
+  b: -5;
+  b: -4;
+  b: -3;
+  b: -2;
+  b: -1;
+}
+
+<===>
+================================================================================
+<===> to_scope/input.scss
+// Overriding the variable inside the loop will not impact the end of the loop.
+$limit: 4;
+
+@for $i from 1 through $limit {
+  $limit: 2;
+  a {
+    b: $limit c $i;
```

Comment:
This is a test case where the property name *isn't* irrelevant, so using a single-letter placeholder does more harm than good. If you write instead:

```suggestion
    limit: $limit;
    i: $i;
```

then it's clear when looking at the output which number is which.